interp/interp.go

```diff
@@ -721,9 +726,23 @@ func fixStdio(interp *Interpreter) {
 	}
 
 	if p = interp.binPkg["os"]; p != nil {
-		p["Stdin"] = reflect.ValueOf(&stdin).Elem()
-		p["Stdout"] = reflect.ValueOf(&stdout).Elem()
-		p["Stderr"] = reflect.ValueOf(&stderr).Elem()
+		if interp.specialStdio {
+			// Inherit streams from interpreter even if they do not have a file descriptor.
+			p["Stdin"] = reflect.ValueOf(&stdin).Elem()
+			p["Stdout"] = reflect.ValueOf(&stdout).Elem()
+			p["Stderr"] = reflect.ValueOf(&stderr).Elem()
+		} else {
```

Comment:
To be honest, I now think that maybe it was a mistake that we allowed that in the past, and we should stop doing it.
But it's not urgent and we can do it later, and the PR fixes the current problem, so ok for now.

I'll approve but I'll let you merge yourself, to be sure you are certain you want to keep going this way.